dojo/tools/cobalt_api/parser.py

```diff
@@ -0,0 +1,180 @@
+            "carried_over",  # Finding from a previous pentest
+            "check_fix",     # Fix for finding is being verified
+            "duplicate",     # Finding is a duplicate within the pentest
+            "invalid",       # Finding is found to be a false positive
+            "need_fix",      # Finding is verified and valid
+            "new",           # The finding is not yet verified by the pentest team
+            "out_of_scope",  # Finding is out of the scope of the pentest
+            "triaging",      # The finding is not yet verified by the pentest team
+            "valid_fix",     # Fix for finding has been varified
+            "wont_fix",      # Risk of finding has been accepted
+        ]
+
+        if resource["state"] in allowed_states:
+            return True
+        else:
+            return False
+
+    def convert_endpoints(self, affected_targets):
+        """Convert Cobalt affected_targets into DefectDojo endpoints"""
+        endpoints = []
+        for affected_target in affected_targets:
+            endpoint = Endpoint.from_uri(affected_target)
+            endpoints.append(endpoint)
+        return endpoints
+
+    def convert_log_timestamp(self, timestamp):
+        """Convert a log entry's timestamp to a DefectDojo date"""
+        date_obj = datetime.strptime(timestamp, "%Y-%m-%dT%H:%M:%S.%fZ")
+        return date_obj.strftime("%Y-%m-%d")
+
+    def convert_severity(self, cobalt_severity):
+        """Convert severity value"""
+        if cobalt_severity == "low":
+            return "Low"
+        elif cobalt_severity == "medium":
+            return "Medium"
+        elif cobalt_severity == "high":
+            return "High"
+        else:
+            return "Info"
+
+    def convert_severity_numerical(self, cobalt_severity):
+        """Convert severity value"""
+        if cobalt_severity == "low":
+            return "S3"
+        elif cobalt_severity == "medium":
+            return "S2"
+        elif cobalt_severity == "high":
+            return "S1"
+        else:
+            return "S4"
+
+    def is_active(self, cobalt_state):
+        return not self.is_mitigated(cobalt_state) \
+            and not self.is_false_p(cobalt_state) \
+            and not self.is_out_of_scope(cobalt_state)
+
+    def is_duplicate(self, cobalt_state):
+        return cobalt_state == "duplicate"
+
+    def is_false_p(self, cobalt_state):
+        return cobalt_state == "invalid"
+
+    def is_mitigated(self, cobalt_state):
+        return cobalt_state == "valid_fix"
+
+    def is_out_of_scope(self, cobalt_state):
+        return cobalt_state == "out_of_scope"
+
+    def is_risk_accepted(self, cobalt_state):
+        return cobalt_state == "wont_fix"
+
+    def is_verified(self, cobalt_state):
+        return cobalt_state != "new" and cobalt_state != "triaging"
+
+    def safe_title(self, title):
+        if len(title) > 255:
+            return title[0:252] + "..."
+        else:
+            return title
```

Comment:
Use the function `shorten` of the package `textwrap` here ( https://docs.python.org/3/library/textwrap.html#textwrap.shorten ).

Reply:
Thanks for the suggestion, removed this helper and added `textwrap.shorten` where this helper used to be called.